src/hm/hmRadio.h

```diff
@@ -147,32 +151,31 @@ class HmRadio {
             bool tx_ok, tx_fail, rx_ready;
             mNrf24.whatHappened(tx_ok, tx_fail, rx_ready);  // resets the IRQ pin to HIGH
             mNrf24.flush_tx();                              // empty TX FIFO
-            //DBGPRINTLN("TX whatHappened Ch" + String(mRfChLst[mTxChIdx]) + " " + String(tx_ok) + String(tx_fail) + String(rx_ready));
 
-            // start listening on the default RX channel
-            mRxChIdx = 0;
+            // start listening
             mNrf24.setChannel(mRfChLst[mRxChIdx]);
             mNrf24.startListening();
 
-            //uint32_t debug_ms = millis();
-            uint16_t cnt = 300; // that is 60 times 5 channels
-            while (0 < cnt--) {
-                uint32_t startMillis = millis();
-                while (millis()-startMillis < 4) {  // listen 4ms to each channel
+            uint32_t startMicros = micros();
+            uint32_t loopMillis = millis();
+            while (millis()-loopMillis < 400) {
+                while (micros()-startMicros < 5110) {  // listen (4088us or?) 5110us to each channel
                     if (mIrqRcvd) {
                         mIrqRcvd = false;
                         if (getReceived()) {        // everything received
-                            //DBGPRINTLN("RX finished Cnt: " + String(300-cnt) + " time used: " + String(millis()-debug_ms)+ " ms");
                             return true;
                         }
                     }
                     yield();
                 }
-                switchRxCh();   // switch to next RX channel
+                // switch to next RX channel
+                startMicros = micros();
+                if(++mRxChIdx >= RF_CHANNELS)
+                    mRxChIdx = 0;
+                mNrf24.setChannel(mRfChLst[mRxChIdx]);
```

Comment:
hier fällt mir auf, dass in `switchRxCh()` bisher `mNrf24.stopListening()` bzw. `mNrf24.startListening()` aufgerufen wurde. 
@beegee3 ist das wirklich möglich ohne diese beiden Aufrufen den Channel zu wechseln?

Reply:
ja, läuft bei rejoe und mir!